src/hooks/useCustomControl.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState, useCallback } from 'react';
+import { ControlProps } from '@jsonforms/core';
+
+function useCustomControl({ data: value, errors, handleChange, path }: ControlProps) {
+	const [isFocused, setFocused] = useState<boolean>(false);
+	const [isDirty, setIsDirty] = useState<boolean>(!!value);
+	const isValid = errors.length === 0 && isDirty;
+
+	const onFocus = useCallback(() => {
+		setFocused(true);
+		setIsDirty(true);
+	}, [path]);
+
+	const onBlur = useCallback(
+		(e) => {
+			setFocused(false);
+			handleChange(path, e.currentTarget.value || '');
+		},
+		[path],
+	);
+
+	const onChange = useCallback(
+		(e) => {
+			handleChange(path, e.currentTarget.value || '');
+		},
+		[path],
+	);
+
+	return { isFocused, isDirty, isValid, onFocus, onBlur, onChange };
+}
+
+export default useCustomControl;
```

Comment:
biedt de AMS CL dit niet? 
Naar mijn smaak mis je nog een level op form niveau, dit doet alleen iets per field en zal ook een andere hook behoeven per type input element (radio, select, etc.) 
Het lijkt me niet de plek om dit in deze package onder te brengen, dit vanuit een CL met form functionaliteit moeten komen, en ofwel hier geimporteerd moeten worden. ofwel en in mijn ogen beter, vanuit de consuming app geinjecteerd moeten worden.

Reply:
This hook is specifically for connecting the control coming from ASC (CL) to JsonForms. I did discuss with one of the core ASC developers whether we should make a dedicated package for these "field wrappers". Optionally we can do so once we have a more complete set. In that case we can move these over to the ASC lerna monorepo and publish it as a standalone package from there. I mentioned this in the vakgroup-frontend meeting today actually.

Reply:
That sounds like a good direction 👍